pkg/controller/che/che_controller.go

```diff
@@ -861,14 +870,18 @@ func (r *ReconcileChe) Reconcile(request reconcile.Request) (reconcile.Result, e
 			return reconcile.Result{}, err
 		}
 	}
-	deployment, _ = r.GetEffectiveDeployment(instance, cheDeployment.Name)
-	actualMemRequest := deployment.Spec.Template.Spec.Containers[0].Resources.Requests[corev1.ResourceMemory]
-	actualMemLimit := deployment.Spec.Template.Spec.Containers[0].Resources.Limits[corev1.ResourceMemory]
+	effectiveCheDeployment, _ = r.GetEffectiveDeployment(instance, cheDeploymentToCreate.Name)
+	actualMemRequest := effectiveCheDeployment.Spec.Template.Spec.Containers[0].Resources.Requests[corev1.ResourceMemory]
+	actualMemLimit := effectiveCheDeployment.Spec.Template.Spec.Containers[0].Resources.Limits[corev1.ResourceMemory]
 	limitStr := actualMemLimit.String()
 	requestStr := actualMemRequest.String()
 	desiredRequest := util.GetValue(instance.Spec.Server.ServerMemoryRequest, deploy.DefaultServerMemoryRequest)
 	desiredLimit := util.GetValue(instance.Spec.Server.ServerMemoryLimit, deploy.DefaultServerMemoryLimit)
-	if desiredRequest != requestStr || desiredLimit != limitStr {
+	desiredImagePullPolicy := util.GetValue(string(instance.Spec.Server.CheImagePullPolicy), deploy.DefaultPullPolicyFromDockerImage(cheImageRepo + ":" + cheImageTag))
+	if desiredRequest != requestStr ||
+		desiredLimit != limitStr ||
+		string(effectiveCheDeployment.Spec.Template.Spec.Containers[0].ImagePullPolicy) != desiredImagePullPolicy ||
+		(r.GetDeploymentEnv(effectiveCheDeployment, "CHE_SELF__SIGNED__CERT") != "") != instance.Spec.Server.SelfSignedCert {
```

Comment:
these last 2 checks are new. Why do we need them?

Reply:
Because these values are used in the Che deployment, so when they are changed, the deployment should be updated and the underlying POD restarted. It was a bug that they were not.

Reply:
+1